packages/mux-player-react/src/index.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect } from "react";
+import "@mux-elements/mux-player";
+import type MuxPlayerElement from "@mux-elements/mux-player";
+import { toNativeProps } from "./common/utils";
+import { useRef } from "react";
+import { useCombinedRefs } from "./useCombinedRefs";
+import useObjectPropEffect from "./useElementPropsEffect";
+
+type ValueOf<T> = T[keyof T];
+
+export type MuxPlayerRefAttributes = MuxPlayerElement;
+
+type VideoApiAttributes = {
+  currentTime: number;
+  volume: number;
+  paused: boolean;
+  src: string | null;
+  poster: string;
+  playbackRate: number;
+  crossOrigin: string;
+  autoPlay: boolean;
+  loop: boolean;
+  muted: boolean;
```

Comment:
should playsinline and preload be added here?

Reply:
For sure. Good callout.

Reply:
actually `preload` should be irrelevant since we rely on hls.js. We should probably remove that from our interface.

Reply:
not always if native hls / m3u8 is used right?

also a good topic for discussion, I think preload should still behave similarly with hls.js

if the user explicitly sets preload="none" I think it should not start loading any segments in hls.js and we should handle this logic in mux-video / playbackcore with setting hls.js configs.